*** Starting uWSGI 2.0.19.1 (64bit) on [Wed Jun 30 10:18:27 2021] ***
compiled with version: 4.8.5 20150623 (Red Hat 4.8.5-44) on 30 June 2021 02:02:53
os: Linux-3.10.0-1160.15.2.el7.x86_64 #1 SMP Wed Feb 3 15:06:38 UTC 2021
nodename: xx-194
machine: x86_64
clock source: unix
detected number of CPU cores: 4
current working directory: /work/django-test/learn
writing pidfile to /work/django-test/learn/uwsgi9090.pid    //pid文件，用于下面的脚本启动、停止该进程
open("/work/django-test/learn/uwsgi9090.pid    //pid文件，用于下面的脚本启动、停止该进程"): No such file or directory [core/utils.c line 3624]
*** Starting uWSGI 2.0.19.1 (64bit) on [Wed Jun 30 10:19:16 2021] ***
compiled with version: 4.8.5 20150623 (Red Hat 4.8.5-44) on 30 June 2021 02:02:53
os: Linux-3.10.0-1160.15.2.el7.x86_64 #1 SMP Wed Feb 3 15:06:38 UTC 2021
nodename: xx-194
machine: x86_64
clock source: unix
detected number of CPU cores: 4
current working directory: /work/django-test/learn
writing pidfile to /work/django-test/learn/uwsgi9090.pid    //pid文件，用于下面的脚本启动、停止该进程
open("/work/django-test/learn/uwsgi9090.pid    //pid文件，用于下面的脚本启动、停止该进程"): No such file or directory [core/utils.c line 3624]
*** Starting uWSGI 2.0.19.1 (64bit) on [Wed Jun 30 10:20:21 2021] ***
compiled with version: 4.8.5 20150623 (Red Hat 4.8.5-44) on 30 June 2021 02:02:53
os: Linux-3.10.0-1160.15.2.el7.x86_64 #1 SMP Wed Feb 3 15:06:38 UTC 2021
nodename: xx-194
machine: x86_64
clock source: unix
detected number of CPU cores: 4
current working directory: /work/django-test/learn
writing pidfile to /work/django-test/learn/uwsgi9090.pid   # //pid文件，用于下面的脚本启动、停止该进程
open("/work/django-test/learn/uwsgi9090.pid   # //pid文件，用于下面的脚本启动、停止该进程"): No such file or directory [core/utils.c line 3624]
*** Starting uWSGI 2.0.19.1 (64bit) on [Wed Jun 30 10:21:14 2021] ***
compiled with version: 4.8.5 20150623 (Red Hat 4.8.5-44) on 30 June 2021 02:02:53
os: Linux-3.10.0-1160.15.2.el7.x86_64 #1 SMP Wed Feb 3 15:06:38 UTC 2021
nodename: xx-194
machine: x86_64
clock source: unix
detected number of CPU cores: 4
current working directory: /work/django-test/learn
writing pidfile to /work/django-test/learn/uwsgi9090.pid
detected binary path: /root/.local/share/virtualenvs/django-test-yalJ4dx_/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 15642
limiting address space of processes...
your process address space limit is 536870912 bytes (512 MB)
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:9090 fd 3
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.6.8 (default, Nov 16 2020, 16:55:22)  [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
*** Python threads support is disabled. You can enable it with --enable-threads ***
Python main interpreter initialized at 0x1822e90
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 296424 bytes (289 KB) for 2 cores
*** Operational MODE: preforking ***
*** no app loaded. going in full dynamic mode ***
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 21497)
spawned uWSGI worker 1 (pid: 21538, cores: 1)
spawned uWSGI worker 2 (pid: 21539, cores: 1)
Traceback (most recent call last):
  File "./learn/wsgi.py", line 16, in <module>
    application = get_wsgi_application()
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/core/wsgi.py", line 12, in get_wsgi_application
    django.setup(set_prefix=False)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/apps/registry.py", line 114, in populate
    app_config.import_models()
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/apps/config.py", line 301, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/lib64/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/contrib/auth/models.py", line 3, in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/contrib/auth/base_user.py", line 48, in <module>
    class AbstractBaseUser(models.Model):
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/models/base.py", line 122, in __new__
    new_class.add_to_class('_meta', Options(meta, app_label))
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/models/base.py", line 326, in add_to_class
    value.contribute_to_class(cls, name)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/models/options.py", line 207, in contribute_to_class
    self.db_table = truncate_name(self.db_table, connection.ops.max_name_length())
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/utils/connection.py", line 15, in __getattr__
    return getattr(self._connections[self._alias], item)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/utils/connection.py", line 62, in __getitem__
    conn = self.create_connection(alias)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/utils.py", line 204, in create_connection
    backend = load_backend(db['ENGINE'])
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/utils.py", line 111, in load_backend
    return import_module('%s.base' % backend_name)
  File "/usr/lib64/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/backends/sqlite3/base.py", line 73, in <module>
    check_sqlite_version()
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/backends/sqlite3/base.py", line 69, in check_sqlite_version
    'SQLite 3.9.0 or later is required (found %s).' % Database.sqlite_version
django.core.exceptions.ImproperlyConfigured: SQLite 3.9.0 or later is required (found 3.7.17).
unable to load app 0 (mountpoint='') (callable not found or import error)
--- no python application found, check your startup logs for errors ---
[pid: 21539|app: -1|req: -1/1] 172.16.7.55 () {48 vars in 1200 bytes} [Wed Jun 30 02:21:38 2021] GET /runoob/ => generated 21 bytes in 2446 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 0)
Traceback (most recent call last):
  File "./learn/wsgi.py", line 16, in <module>
    application = get_wsgi_application()
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/core/wsgi.py", line 12, in get_wsgi_application
    django.setup(set_prefix=False)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/__init__.py", line 24, in setup
    apps.populate(settings.INSTALLED_APPS)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/apps/registry.py", line 114, in populate
    app_config.import_models()
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/apps/config.py", line 301, in import_models
    self.models_module = import_module(models_module_name)
  File "/usr/lib64/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/contrib/auth/models.py", line 3, in <module>
    from django.contrib.auth.base_user import AbstractBaseUser, BaseUserManager
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/contrib/auth/base_user.py", line 48, in <module>
    class AbstractBaseUser(models.Model):
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/models/base.py", line 122, in __new__
    new_class.add_to_class('_meta', Options(meta, app_label))
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/models/base.py", line 326, in add_to_class
    value.contribute_to_class(cls, name)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/models/options.py", line 207, in contribute_to_class
    self.db_table = truncate_name(self.db_table, connection.ops.max_name_length())
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/utils/connection.py", line 15, in __getattr__
    return getattr(self._connections[self._alias], item)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/utils/connection.py", line 62, in __getitem__
    conn = self.create_connection(alias)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/utils.py", line 204, in create_connection
    backend = load_backend(db['ENGINE'])
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/utils.py", line 111, in load_backend
    return import_module('%s.base' % backend_name)
  File "/usr/lib64/python3.6/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/backends/sqlite3/base.py", line 73, in <module>
    check_sqlite_version()
  File "/root/.local/share/virtualenvs/django-test-yalJ4dx_/lib/python3.6/site-packages/django/db/backends/sqlite3/base.py", line 69, in check_sqlite_version
    'SQLite 3.9.0 or later is required (found %s).' % Database.sqlite_version
django.core.exceptions.ImproperlyConfigured: SQLite 3.9.0 or later is required (found 3.7.17).
unable to load app 0 (mountpoint='') (callable not found or import error)
--- no python application found, check your startup logs for errors ---
[pid: 21538|app: -1|req: -1/2] 172.16.7.55 () {46 vars in 1167 bytes} [Wed Jun 30 02:22:04 2021] GET /hello/ => generated 21 bytes in 622 msecs (HTTP/1.1 500) 2 headers in 83 bytes (0 switches on core 0)
...brutally killing workers...
worker 1 buried after 1 seconds
worker 2 buried after 1 seconds
binary reloading uWSGI...
chdir() to /work/django-test/learn
closing all non-uwsgi socket fds > 2 (max_fd = 1024)...
found fd 3 mapped to socket 0 (127.0.0.1:9090)
running /root/.local/share/virtualenvs/django-test-yalJ4dx_/bin/uwsgi
[uWSGI] getting INI configuration from uwsgi9090.ini
*** Starting uWSGI 2.0.19.1 (64bit) on [Wed Jun 30 10:35:30 2021] ***
compiled with version: 4.8.5 20150623 (Red Hat 4.8.5-44) on 30 June 2021 02:02:53
os: Linux-3.10.0-1160.15.2.el7.x86_64 #1 SMP Wed Feb 3 15:06:38 UTC 2021
nodename: xx-194
machine: x86_64
clock source: unix
detected number of CPU cores: 4
current working directory: /work/django-test/learn
detected binary path: /root/.local/share/virtualenvs/django-test-yalJ4dx_/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 15642
limiting address space of processes...
your process address space limit is 536870912 bytes (512 MB)
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 inherited INET address 127.0.0.1:9090 fd 3
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.6.8 (default, Nov 16 2020, 16:55:22)  [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
*** Python threads support is disabled. You can enable it with --enable-threads ***
Python main interpreter initialized at 0x19f17f0
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 296424 bytes (289 KB) for 2 cores
*** Operational MODE: preforking ***
*** no app loaded. going in full dynamic mode ***
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
gracefully (RE)spawned uWSGI master process (pid: 21497)
spawned uWSGI worker 1 (pid: 43024, cores: 1)
spawned uWSGI worker 2 (pid: 43025, cores: 1)
...brutally killing workers...
worker 1 buried after 1 seconds
worker 2 buried after 1 seconds
binary reloading uWSGI...
chdir() to /work/django-test/learn
closing all non-uwsgi socket fds > 2 (max_fd = 1024)...
found fd 3 mapped to socket 0 (127.0.0.1:9090)
running /root/.local/share/virtualenvs/django-test-yalJ4dx_/bin/uwsgi
[uWSGI] getting INI configuration from uwsgi9090.ini
*** Starting uWSGI 2.0.19.1 (64bit) on [Wed Jun 30 10:35:41 2021] ***
compiled with version: 4.8.5 20150623 (Red Hat 4.8.5-44) on 30 June 2021 02:02:53
os: Linux-3.10.0-1160.15.2.el7.x86_64 #1 SMP Wed Feb 3 15:06:38 UTC 2021
nodename: xx-194
machine: x86_64
clock source: unix
detected number of CPU cores: 4
current working directory: /work/django-test/learn
detected binary path: /root/.local/share/virtualenvs/django-test-yalJ4dx_/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 15642
limiting address space of processes...
your process address space limit is 536870912 bytes (512 MB)
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 inherited INET address 127.0.0.1:9090 fd 3
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.6.8 (default, Nov 16 2020, 16:55:22)  [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
*** Python threads support is disabled. You can enable it with --enable-threads ***
Python main interpreter initialized at 0x105f7f0
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 296424 bytes (289 KB) for 2 cores
*** Operational MODE: preforking ***
*** no app loaded. going in full dynamic mode ***
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
gracefully (RE)spawned uWSGI master process (pid: 21497)
spawned uWSGI worker 1 (pid: 43327, cores: 1)
spawned uWSGI worker 2 (pid: 43328, cores: 1)
Wed Jun 30 10:35:59 2021 - uWSGI worker 1 screams: UAAAAAAH my master disconnected: i will kill myself !!!
Wed Jun 30 10:35:59 2021 - uWSGI worker 2 screams: UAAAAAAH my master disconnected: i will kill myself !!!
*** Starting uWSGI 2.0.19.1 (64bit) on [Wed Jun 30 10:36:20 2021] ***
compiled with version: 4.8.5 20150623 (Red Hat 4.8.5-44) on 30 June 2021 02:02:53
os: Linux-3.10.0-1160.15.2.el7.x86_64 #1 SMP Wed Feb 3 15:06:38 UTC 2021
nodename: xx-194
machine: x86_64
clock source: unix
detected number of CPU cores: 4
current working directory: /work/django-test/learn
writing pidfile to /work/django-test/learn/uwsgi9090.pid
detected binary path: /root/.local/share/virtualenvs/django-test-yalJ4dx_/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 15642
limiting address space of processes...
your process address space limit is 536870912 bytes (512 MB)
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:9090 fd 3
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.6.8 (default, Nov 16 2020, 16:55:22)  [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
*** Python threads support is disabled. You can enable it with --enable-threads ***
Python main interpreter initialized at 0x18e9e90
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 296424 bytes (289 KB) for 2 cores
*** Operational MODE: preforking ***
*** no app loaded. going in full dynamic mode ***
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 44313)
spawned uWSGI worker 1 (pid: 44324, cores: 1)
spawned uWSGI worker 2 (pid: 44325, cores: 1)
WSGI app 0 (mountpoint='') ready in 1 seconds on interpreter 0x18e9e90 pid: 44325 (default app)
[pid: 44325|app: 0|req: 1/1] 172.16.7.55 () {48 vars in 1198 bytes} [Wed Jun 30 02:36:43 2021] GET /hello/ => generated 14 bytes in 1113 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
WSGI app 0 (mountpoint='') ready in 1 seconds on interpreter 0x18e9e90 pid: 44324 (default app)
[pid: 44324|app: 0|req: 1/2] 172.16.7.55 () {46 vars in 1169 bytes} [Wed Jun 30 02:36:57 2021] GET /runoob/ => generated 72 bytes in 852 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 44325|app: 0|req: 2/3] 172.16.7.55 () {48 vars in 1200 bytes} [Wed Jun 30 02:37:10 2021] GET /runoob/ => generated 72 bytes in 33 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 44324|app: 0|req: 2/4] 172.16.7.55 () {48 vars in 1200 bytes} [Wed Jun 30 02:37:15 2021] GET /runoob/ => generated 72 bytes in 4 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 44324|app: 0|req: 3/5] 172.16.7.55 () {48 vars in 1200 bytes} [Wed Jun 30 02:37:16 2021] GET /runoob/ => generated 72 bytes in 4 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 44325|app: 0|req: 3/6] 172.16.7.55 () {48 vars in 1200 bytes} [Wed Jun 30 02:37:17 2021] GET /runoob/ => generated 72 bytes in 4 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 44324|app: 0|req: 4/7] 172.16.7.55 () {48 vars in 1200 bytes} [Wed Jun 30 02:37:17 2021] GET /runoob/ => generated 72 bytes in 4 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 44324|app: 0|req: 5/8] 172.16.7.55 () {48 vars in 1200 bytes} [Wed Jun 30 02:37:18 2021] GET /runoob/ => generated 72 bytes in 4 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 44324|app: 0|req: 6/9] 172.16.7.55 () {46 vars in 1167 bytes} [Wed Jun 30 02:38:25 2021] GET /hello/ => generated 14 bytes in 2 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 44324|app: 0|req: 7/10] 172.16.7.55 () {48 vars in 1200 bytes} [Wed Jun 30 02:38:31 2021] GET /runoob/ => generated 72 bytes in 4 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
SIGINT/SIGQUIT received...killing workers...
worker 1 buried after 1 seconds
worker 2 buried after 1 seconds
goodbye to uWSGI.
VACUUM: pidfile removed.
*** Starting uWSGI 2.0.19.1 (64bit) on [Wed Jun 30 10:39:57 2021] ***
compiled with version: 4.8.5 20150623 (Red Hat 4.8.5-44) on 30 June 2021 02:02:53
os: Linux-3.10.0-1160.15.2.el7.x86_64 #1 SMP Wed Feb 3 15:06:38 UTC 2021
nodename: xx-194
machine: x86_64
clock source: unix
detected number of CPU cores: 4
current working directory: /work/django-test/learn
writing pidfile to /work/django-test/learn/uwsgi9090.pid
detected binary path: /root/.local/share/virtualenvs/django-test-yalJ4dx_/bin/uwsgi
!!! no internal routing support, rebuild with pcre support !!!
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your processes number limit is 15642
limiting address space of processes...
your process address space limit is 536870912 bytes (512 MB)
your memory page size is 4096 bytes
detected max file descriptor number: 1024
lock engine: pthread robust mutexes
thunder lock: disabled (you can enable it with --thunder-lock)
uwsgi socket 0 bound to TCP address 127.0.0.1:9090 fd 3
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
Python version: 3.6.8 (default, Nov 16 2020, 16:55:22)  [GCC 4.8.5 20150623 (Red Hat 4.8.5-44)]
*** Python threads support is disabled. You can enable it with --enable-threads ***
Python main interpreter initialized at 0x20abe90
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
your server socket listen backlog is limited to 100 connections
your mercy for graceful operations on workers is 60 seconds
mapped 296424 bytes (289 KB) for 2 cores
*** Operational MODE: preforking ***
*** no app loaded. going in full dynamic mode ***
uWSGI running as root, you can use --uid/--gid/--chroot options
*** WARNING: you are running uWSGI as root !!! (use the --uid flag) *** 
*** uWSGI is running in multiple interpreter mode ***
spawned uWSGI master process (pid: 49716)
spawned uWSGI worker 1 (pid: 49726, cores: 1)
spawned uWSGI worker 2 (pid: 49727, cores: 1)
WSGI app 0 (mountpoint='') ready in 1 seconds on interpreter 0x20abe90 pid: 49727 (default app)
[pid: 49727|app: 0|req: 1/1] 172.16.7.55 () {48 vars in 1200 bytes} [Wed Jun 30 02:40:28 2021] GET /runoob/ => generated 72 bytes in 747 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 49727|app: 0|req: 2/2] 172.16.7.55 () {48 vars in 1200 bytes} [Wed Jun 30 03:11:26 2021] GET /runoob/ => generated 72 bytes in 4 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 49727|app: 0|req: 3/3] 172.16.7.55 () {46 vars in 1192 bytes} [Wed Jun 30 03:11:35 2021] GET /runoob/?name=234234 => generated 74 bytes in 5 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 49727|app: 0|req: 4/4] 172.16.7.55 () {48 vars in 1223 bytes} [Wed Jun 30 03:11:38 2021] GET /runoob/?name=234234 => generated 74 bytes in 5 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
WSGI app 0 (mountpoint='') ready in 3 seconds on interpreter 0x20abe90 pid: 49726 (default app)
[pid: 49726|app: 0|req: 1/5] 172.16.7.55 () {46 vars in 958 bytes} [Thu Jul  1 06:54:12 2021] GET / => generated 14 bytes in 3147 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 49727|app: 0|req: 5/6] 172.16.7.55 () {48 vars in 989 bytes} [Thu Jul  1 06:59:01 2021] GET / => generated 14 bytes in 2 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 49726|app: 0|req: 2/7] 172.16.7.55 () {46 vars in 971 bytes} [Thu Jul  1 07:06:05 2021] GET /runoob/ => generated 72 bytes in 196 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 49727|app: 0|req: 6/8] 172.16.7.55 () {46 vars in 957 bytes} [Thu Jul  1 07:06:12 2021] GET / => generated 14 bytes in 2 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
Not Found: /sdf
[pid: 49727|app: 0|req: 7/9] 172.16.7.55 () {46 vars in 963 bytes} [Thu Jul  1 07:06:44 2021] GET /sdf => generated 2297 bytes in 64 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /index
[pid: 49727|app: 0|req: 8/10] 172.16.7.55 () {46 vars in 967 bytes} [Thu Jul  1 07:13:22 2021] GET /index => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /login
[pid: 49727|app: 0|req: 9/11] 172.16.7.55 () {46 vars in 967 bytes} [Thu Jul  1 07:13:28 2021] GET /login => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /index
[pid: 49726|app: 0|req: 3/12] 172.16.7.55 () {46 vars in 967 bytes} [Thu Jul  1 07:14:34 2021] GET /index => generated 2303 bytes in 17 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /login
[pid: 49727|app: 0|req: 10/13] 172.16.7.55 () {46 vars in 967 bytes} [Thu Jul  1 07:14:44 2021] GET /login => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
[pid: 49726|app: 0|req: 4/14] 172.16.7.55 () {46 vars in 969 bytes} [Thu Jul  1 07:15:07 2021] GET /hello/ => generated 14 bytes in 2 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 49727|app: 0|req: 11/15] 172.16.7.55 () {46 vars in 971 bytes} [Thu Jul  1 07:15:14 2021] GET /runoob/ => generated 72 bytes in 4 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 49727|app: 0|req: 12/16] 172.16.7.55 () {48 vars in 1002 bytes} [Thu Jul  1 07:16:00 2021] GET /runoob/ => generated 72 bytes in 4 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
Not Found: /login
[pid: 49726|app: 0|req: 5/17] 172.16.7.55 () {46 vars in 967 bytes} [Thu Jul  1 07:16:04 2021] GET /login => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /login
[pid: 49727|app: 0|req: 13/18] 172.16.7.55 () {46 vars in 967 bytes} [Thu Jul  1 07:16:05 2021] GET /login => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /login
[pid: 49727|app: 0|req: 14/19] 172.16.7.55 () {48 vars in 998 bytes} [Thu Jul  1 07:17:20 2021] GET /login => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /login
[pid: 49727|app: 0|req: 15/20] 172.16.7.55 () {48 vars in 998 bytes} [Thu Jul  1 07:17:24 2021] GET /login => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /login
[pid: 49727|app: 0|req: 16/21] 172.16.7.55 () {48 vars in 998 bytes} [Thu Jul  1 07:18:42 2021] GET /login => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /login
[pid: 49727|app: 0|req: 17/22] 172.16.7.55 () {48 vars in 998 bytes} [Thu Jul  1 07:21:16 2021] GET /login => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /login
[pid: 49726|app: 0|req: 6/23] 172.16.7.55 () {48 vars in 998 bytes} [Thu Jul  1 07:25:40 2021] GET /login => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /index
[pid: 49726|app: 0|req: 7/24] 172.16.7.55 () {46 vars in 967 bytes} [Thu Jul  1 07:25:45 2021] GET /index => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /index
[pid: 49727|app: 0|req: 18/25] 172.16.7.55 () {48 vars in 998 bytes} [Thu Jul  1 07:26:49 2021] GET /index => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /index
[pid: 49726|app: 0|req: 8/26] 172.16.7.55 () {48 vars in 998 bytes} [Thu Jul  1 07:28:38 2021] GET /index => generated 2303 bytes in 12 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
[pid: 49727|app: 0|req: 19/27] 172.16.7.55 () {44 vars in 786 bytes} [Thu Jul  1 07:28:55 2021] GET / => generated 14 bytes in 2 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
Not Found: /favicon.ico
[pid: 49727|app: 0|req: 20/28] 172.16.7.55 () {44 vars in 743 bytes} [Thu Jul  1 07:28:56 2021] GET /favicon.ico => generated 2321 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
[pid: 49727|app: 0|req: 21/29] 172.16.7.55 () {44 vars in 796 bytes} [Thu Jul  1 07:29:13 2021] GET /hello => generated 0 bytes in 1 msecs (HTTP/1.1 301) 5 headers in 175 bytes (1 switches on core 0)
[pid: 49727|app: 0|req: 22/30] 172.16.7.55 () {44 vars in 798 bytes} [Thu Jul  1 07:29:13 2021] GET /hello/ => generated 14 bytes in 1 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
[pid: 49727|app: 0|req: 23/31] 172.16.7.55 () {44 vars in 798 bytes} [Thu Jul  1 07:29:34 2021] GET /runoob => generated 0 bytes in 2 msecs (HTTP/1.1 301) 5 headers in 176 bytes (1 switches on core 0)
[pid: 49727|app: 0|req: 24/32] 172.16.7.55 () {44 vars in 800 bytes} [Thu Jul  1 07:29:34 2021] GET /runoob/ => generated 72 bytes in 3 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
Not Found: /runoob/>234234
[pid: 49727|app: 0|req: 25/33] 172.16.7.55 () {44 vars in 816 bytes} [Thu Jul  1 07:29:38 2021] GET /runoob/%3E234234 => generated 2338 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
[pid: 49726|app: 0|req: 9/34] 172.16.7.55 () {44 vars in 823 bytes} [Thu Jul  1 07:29:47 2021] GET /runoob/?name=882342 => generated 74 bytes in 4 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
Not Found: /login
[pid: 49727|app: 0|req: 26/35] 172.16.7.55 () {44 vars in 796 bytes} [Thu Jul  1 07:29:52 2021] GET /login => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /login
[pid: 49727|app: 0|req: 27/36] 172.16.7.55 () {46 vars in 827 bytes} [Thu Jul  1 07:30:52 2021] GET /login => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /login
[pid: 49727|app: 0|req: 28/37] 172.16.7.55 () {46 vars in 827 bytes} [Thu Jul  1 07:30:54 2021] GET /login => generated 2303 bytes in 10 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /login
[pid: 49727|app: 0|req: 29/38] 172.16.7.55 () {46 vars in 827 bytes} [Thu Jul  1 07:30:55 2021] GET /login => generated 2303 bytes in 11 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /login
[pid: 49727|app: 0|req: 30/39] 172.16.7.55 () {46 vars in 827 bytes} [Thu Jul  1 07:30:57 2021] GET /login => generated 2303 bytes in 10 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
Not Found: /login
[pid: 49726|app: 0|req: 10/40] 172.16.7.55 () {46 vars in 827 bytes} [Thu Jul  1 07:31:00 2021] GET /login => generated 2303 bytes in 15 msecs (HTTP/1.1 404) 5 headers in 159 bytes (1 switches on core 0)
[pid: 49727|app: 0|req: 31/41] 172.16.7.55 () {44 vars in 786 bytes} [Thu Jul  1 07:31:01 2021] GET / => generated 14 bytes in 2 msecs (HTTP/1.1 200) 5 headers in 165 bytes (1 switches on core 0)
SIGINT/SIGQUIT received...killing workers...
worker 1 buried after 1 seconds
worker 2 buried after 1 seconds
goodbye to uWSGI.
VACUUM: pidfile removed.
